airflow/jobs/scheduler_job_runner.py

```diff
@@ -1308,8 +1308,18 @@ def _create_dag_runs_dataset_triggered(
                     DatasetDagRunQueue.target_dag_id == dag_run.dag_id
                 ).delete()
 
-    def _should_update_dag_next_dagruns(self, dag, dag_model: DagModel, total_active_runs: int) -> bool:
+    def _should_update_dag_next_dagruns(
+        self, dag: DAG, dag_model: DagModel, session: Session, total_active_runs: int = -1
+    ) -> bool:
         """Check if the dag's next_dagruns_create_after should be updated."""
+        # If dag has no interval or timetable then skip save runtime
+        if not dag.schedule_interval and isinstance(dag.timetable, NullTimetable):
+            return False
```

Comment:
Multiple issues and questions I’m going to be lazy and merge

1. The `schedule_interval` is simply a string describing the timetable, checking against it is not meaningful.
2. The argument is very weird, especially with the `-1` default (and why is that needed really?)
3. How would this work if a user subclass NullTimetable?
4. A flag on the timetable class would be preferred over `isinstance`. See how things like `periodic` are interfaced.

Reply:
Thanks for your input!
1. Any idea what is the best case to check if a dag is only triggered and not scheduled? I was not finding a nicer way to check for this.
2. I wanted to catch the use case of the function _create_dag_runs were the number of active dags is increase outside of the function.
3. Why should a user subclass NullTimetable. I did not get why NullTimetable exists and not setting this to None. Can you help me to understand this? i though NullTimeTable is used if no timetable exists.
4. Will check this out.

Reply:
1. Do you mean if a DAG _can only be manually triggered_? That would be `dag.timetable.can_run`.
2. Not sure I’m getting what you mean on this one.
3. Using `None` would make other parts of the code a lot more complicated because everything everywhere needs to check for None. This is called the [null object pattern](https://en.wikipedia.org/wiki/Null_object_pattern).

Reply:
Add a new attribute that works for this. Checking for class identity should generally be avoided for [polymorphism](https://en.wikipedia.org/wiki/Polymorphism_%28computer_science%29), which flags like `can_run` provide.

Reply:
Okay, thanks for the feedback. I am thinking about how to apply the flag w/o adding more complexity in attributes. Do you think rather (1) we should add a new attribute just for this use case to the `Timetable` class? I was thinking but can not imagine a good self-descriptive name other than `can run`. So (2) would it be probably meaningful to change the default of `can_run`only on `NullTimetable`and `OnceTimetable`? Because actually when taking a look to the code it is rather mis-placed in `ContinuousTimetable` - but I am not sure what other side effects might be?

Reply:
Mhm, the more I think of it I assume `can_run` is not the right word, If I take a look to the code it is rather matching to the meaning of `is_scheduled`? Because otherwise would mean like `only_manual_triggered`?

Reply:
Maybe `schedulable` or `can_be_scheduled` might make sense? Or (if _schedule_ is ambiguous) `only_manually_triggered` can work as well. The flag is internal to Airflow (not documented for end users to use) so anything that’s sufficiently descriptive should be OK.